### PR TITLE
Shadows add middle lod

### PR DESCRIPTION
**Pull request by mchaptel**
adds 2 extra maps for more detail in shadows at lesser cost.

### CODE REVIEW

File: TESReloaded/Core/ShadowManager.cpp [hunk 1]
```diff
@@ -487,22 +506,6 @@ void ShadowManager::RenderShadowMaps() {
 	IDirect3DSurface9* DepthSurface = NULL;
 	IDirect3DSurface9* RenderSurface = NULL;
 	D3DVIEWPORT9 viewport;
-/*	if(RenderStateSettings == nullptr){
```
Comment (llde) [inline TESReloaded/Core/ShadowManager.cpp:490]: Don't remove this

File: TESReloaded/Core/ShadowManager.cpp [hunk 2]
```diff
@@ -14,11 +17,22 @@ void ShadowManager::Initialize() {
...
+	TheShaderManager->ShaderConst.ShadowMap.ShadowMapRadius.w = ShadowsExteriors->ShadowMapRadius[3];
+
+	ShadowMapSize = ShadowsExteriors->ShadowMapSize[0];
     TheShaderManager->CreateFrameVertex(ShadowMapSize, ShadowMapSize, &TheShadowManager->BlurShadowVertex[0]);
-    ShadowMapSize = ShadowsExteriors->ShadowMapSize[1];
-    TheShaderManager->CreateFrameVertex(ShadowMapSize, ShadowMapSize, &TheShadowManager->BlurShadowVertex[1]);
+	ShadowMapSize = ShadowsExteriors->ShadowMapSize[1];
+	TheShaderManager->CreateFrameVertex(ShadowMapSize, ShadowMapSize, &TheShadowManager->BlurShadowVertex[1]);
+	ShadowMapSize = ShadowsExteriors->ShadowMapSize[2];
+	TheShaderManager->CreateFrameVertex(ShadowMapSize, ShadowMapSize, &TheShadowManager->BlurShadowVertex[2]);
```
Comment (llde) [inline TESReloaded/Core/ShadowManager.cpp:33]: the frame vertex can be the same. No need to replicate it 4 times

**Review by llde — CHANGES_REQUESTED** [review]

File: TESReloaded/Core/ShadowManager.cpp (hunk 1 above)
Comment (mchaptel) [reply TESReloaded/Core/ShadowManager.cpp:490]: should we uncomment it?

File: TESReloaded/Core/ShadowManager.cpp [hunk 3]
```diff
@@ -14,11 +17,22 @@ void ShadowManager::Initialize() {
...
+	ShadowMapSize = ShadowsExteriors->ShadowMapSize[3];
```
Comment (llde) [inline TESReloaded/Core/ShadowManager.cpp:34]: you can create these FrameVertex in the loop

File: TESReloaded/Core/ShadowManager.cpp (hunk 1 above)
Comment (llde) [reply TESReloaded/Core/ShadowManager.cpp:490]: no keep commented, but don't remove it

File: TESReloaded/Core/ShadowManager.cpp (hunk 3 above)
Comment (mchaptel) [reply TESReloaded/Core/ShadowManager.cpp:34]: Like this? 

(line 36)
```
	for (int i = 0; i < 5; i++) {
		ShadowMapSize = ShadowsExteriors->ShadowMapSize[i];
		TheShadowManager->ShadowMapViewPort[i] = { 0, 0, ShadowMapSize, ShadowMapSize, 0.0f, 1.0f };
        TheShadowManager->ShadowMapInverseResolution[i] = 1.0f / (float) ShadowMapSize;
		TheShaderManager->CreateFrameVertex(ShadowMapSize, ShadowMapSize, &TheShadowManager->BlurShadowVertex[i]);
	}
```

**Pull request merged**